### DISCUSSION TITLE
Watchdogs

### DISCUSSION THREAD

**Discussion by Usioumeo** [ch32-rs/ch32-hal#38] [Q&A]
Anyone knows how to correctly set, and modify watchdogs?
Because every 240 seconds my code halts on a ch32v305, and only by pushing the reset button it restarts.
Searching around I've found that this MCU contains two watchdogs: WWDG and IWDG, What is the correct way to set them?

**Comment by andelf** [comment]
There's an AWU that can be used as a wakeup source.
Recently I'm investigate the time driver issues.

"IWDG" is OK.

**Comment by cvbni** [comment]
Does IWDG work now?
My ch32v307 is easy to be interfered with and crashes, for example, if you plug and unplug the serial port pins, it may crash.
I wish there were Watchdogs that could get it to reboot  after a while.

**Comment by Usioumeo** [comment]
This was solved by disabling native atomics.
If someone wants to read more: [qingke issue](https://github.com/ch32-rs/qingke/issues/9) [ch32-hal issue](https://github.com/ch32-rs/ch32-hal/pull/61)
